src/order/directed.lean

```diff
@@ -47,21 +48,60 @@ theorem directed.mono_comp {ι} {rb : β → β → Prop} {g : α → β} {f : 
   directed rb (g ∘ f) :=
 directed_comp.2 $ hf.mono hg
 
-/-- A monotone function on a sup-semilattice is directed. -/
-lemma directed_of_sup [semilattice_sup α] {f : α → β} {r : β → β → Prop}
-  (H : ∀ ⦃i j⦄, i ≤ j → r (f i) (f j)) : directed r f :=
-λ a b, ⟨a ⊔ b, H le_sup_left, H le_sup_right⟩
-
-/-- An antimonotone function on an inf-semilattice is directed. -/
-lemma directed_of_inf [semilattice_inf α] {r : β → β → Prop} {f : α → β}
-  (hf : ∀a₁ a₂, a₁ ≤ a₂ → r (f a₂) (f a₁)) : directed r f :=
-assume x y, ⟨x ⊓ y, hf _ _ inf_le_left, hf _ _ inf_le_right⟩
-
 /-- A `preorder` is a `directed_order` if for any two elements `i`, `j`
 there is an element `k` such that `i ≤ k` and `j ≤ k`. -/
 class directed_order (α : Type u) extends preorder α :=
 (directed : ∀ i j : α, ∃ k, i ≤ k ∧ j ≤ k)
 
+/-- A `preorder` is a `anti_directed_order` if for any two elements `i`, `j`
+there is an element `k` such that `k ≤ i` and `k ≤ j`. -/
+class anti_directed_order (α : Type u) extends preorder α :=
+(directed : ∀ i j : α, ∃ k, k ≤ i ∧ k ≤ j)
```

Comment:
@b-mehta, does this terminology seem ok to you? "upward-directed" and "downward-directed" are another option for the two names, or perhaps even `sup_directed` and `inf_directed`?

Reply:
I think your alternatives are slightly nicer but there's nothing wrong with the original - perhaps ask on Zulip? As long as the names aren't misleading I'm probably pretty happy with whatever you pick.